Documentation/disaster-recovery.md

```diff
@@ -106,3 +106,102 @@ kubectl create -f operator.yaml
 ```
 
 The operator will automatically add more mons to increase the quorum size again, depending on the `monCount`.
+
+# Bring orphaned rook/ceph cluster back to Kubernetes
+
+## What this section trying to resolve
+
+1. You had a running rook/ceph cluster, suddenly your Kubernetes environment exploded, you have to start a new Kubernetes environment and put your existing rook/ceph cluster back.
+2. You are migrating your existing rook/ceph cluster to a new Kubernetes environment, downtime can be tolerated.
+
+In author's situation,  `etcd` data of running Kubernetes cluster is nuked and have no backup, all OSDs are using bluestore backend.
+
+### Prerequisites
+
+1. A working Kubernetes cluster without rook
+2. Previous rook/ceph cluster is intact, which means you have at least one ceph mon data intact, and all your OSD data is intact.
+
+Tested in `rook/ceph:v1.0.3` and `ceph/ceph:v14.2.1-20190430`.
+
+### TL;DR for Steps below
+1. Start a new and clean rook cluster, with old `CephCluster` & friends. Shut it down when it seems working(as a brand new cluster).
+3. Replace ceph-mon data with old one. Fix `fsid` in rook. Fix monmap. Fix mon auth. Disable auth.
+4. Spin it up, watch it resurrect. Fix admin auth key.
+5. Shut it down again. Enable auth. Spin it up.
+
+HOORAY!
+
+### Steps
+
+0. Assuming your old Kubernetes cluster is completely torned down, and your new Kubernetes cluster is up and running, without rook.
+1. Backup `/var/lib/rook` in all your rook nodes. Backups will be used later.
+2. Pick a `/var/lib/rook/rook-ceph/rook-ceph.config`  from any node and get your old cluster `fsid` from its content.
+3. Remove `/var/lib/rook` in all your rook nodes.
+4. Install rook in your new Kubernetes cluster.
+5. Prepare identical `CephCluster`  descriptors, especially identical `spec.storage.config` and `spec.storage.nodes`,  except `mon.count`, which sets to `1`. Post them to your new Kubernetes cluster.
+6. Prepare identical `CephFilesystem` & ... etc descriptors (if any). Post them to your new Kubernetes cluster too.
+7. Run `kubectl logs -f rook-ceph-operator-xxxxxxxxxx` and wait till all the things are settled.
```

Comment:
This could be clearer. "Watch the operator logs with `kubectl logs --follow rook-ceph-operator-xxxxxxx`, and wait until the orchestration has settled." Also include which Ceph daemons the user can expect to see running at this time.

Reply:
fixed